org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/embedder/MavenImpl.java

```diff
@@ -188,7 +191,8 @@ public class MavenImpl implements IMaven, IMavenConfigurationChangeListener, Clo
 
   private final ConverterLookup converterLookup = new DefaultConverterLookup();
 
-  private final List<ISettingsChangeListener> settingsListeners = new ArrayList<>();
+  @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
```

Comment:
`cardinality = ReferenceCardinality.MULTIPLE` is actually not necessary isn't it? (But I'm fine with making this explicit.
IIRC with the dynamic-policy new/removed service implementations are added/removed from the list instead of creating a new list, aren't they?

Reply:
> `cardinality = ReferenceCardinality.MULTIPLE` is actually not necessary isn't it?

maybe I just always add it to make it clear :-)